firmware-management/updated/init.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * This module contains initialisation functions and state.
+ */
+
+#include "init.h"
+
+#include <systemd/sd-daemon.h>
+
+#include <unistd.h>
+
+
+namespace updated {
+    namespace init {
+        Status initialise()
+        {
+            return Status::Started;
+        }
+
+        // Notify the init system that we started successfully
+        void notify_start(const Status startup_status)
+        {
+            switch(startup_status)
+            {
+                case Status::Started:
+                    sd_notify(0, "READY=1");
+                    break;
+
+                case Status::FailedToStart:
+                    sd_notify(0, "STATUS=Failed to start\n");
+                    break;
+
+                default:
```

Comment:
I don't think we want a `default` case here. Since this is a switch over an enum, if we leave out the `default` case then `-Wswitch` will cause warnings/errors when we miss out a case for one of the enum values.

Also, is doing nothing the best thing to do if `startup_status` is out of range? I suggest we panic or assert when `startup_status` is out of range - that would indicate a programmer error.

Reply:
Okay, we don't need a default case here. However, I'm not sure we need to assert, as this function takes an `updated::init::Status` which is a strongly typed `enum` so I'm not sure it's possible for `startup_status` to be out of range.

Reply:
Yep, you're right, putting an out-of-range value into an enum is undefined behaviour so the compiler would be free to assume it never happens and not generate any code for the out-of-range case.

https://github.com/cplusplus/draft/commit/b416ec442dad103d57661007030bca096bd0fd0f